Library/Homebrew/brew.sh

```diff
@@ -350,17 +350,23 @@ else
   curl_name_and_version="${curl_version_output%% (*}"
   if [[ $(numeric "${curl_name_and_version##* }") -lt $(numeric "$HOMEBREW_MINIMUM_CURL_VERSION") ]]
   then
-    if [[ -z $HOMEBREW_CURL_PATH ]]; then
+      message="Your cURL version is too old.
+Minimum required version: ${HOMEBREW_MINIMUM_CURL_VERSION}
+Your cURL version: ${curl_name_and_version##* }
+Your cURL executable: $(type -p $HOMEBREW_CURL)
+To point Homebrew to cURL ${HOMEBREW_MINIMUM_CURL_VERSION} or newer,
+enable developer mode by setting HOMEBREW_DEVELOPER to 1,
+then set HOMEBREW_CURL_PATH to the location of cURL executable."
```

Comment:
This is not something we should be signposting to users that haven't already set `HOMEBREW_DEVELOPER`. I don't think it's worth customising the message for the latter given they are maintainers. Let's direct them to install a newer `curl` in the location we check instead.

Reply:
> Let's direct them to install a newer curl in the location we check instead.

In the standard (non-developer) mode, Homebrew uses/tests system cURL and ignores any other locations. So, if system cURL is out-of-date and we don't mention the developer mode, all we can say is "please update your system cURL".

Alternatively, we could change the behavior and not require `HOMEBREW_DEVELOPER` for `HOMEBREW_CURL_PATH`. In that case we could say "Please update your system cURL or point Homebrew to cURL ${HOMEBREW_MINIMUM_CURL_VERSION} or newer via the HOMEBREW_CURL_PATH variable".

Reply:
> So, if system cURL is out-of-date and we don't mention the developer mode, all we can say is "please update your system cURL".

Yup, sounds good.



> Alternatively, we could change the behavior and not require `HOMEBREW_DEVELOPER` for `HOMEBREW_CURL_PATH`. In that case we could say "Please update your system cURL or point Homebrew to cURL ${HOMEBREW_MINIMUM_CURL_VERSION} or newer via the HOMEBREW_CURL_PATH variable".

I'd rather we require `HOMEBREW_DEVELOPER` (and not tell users) until we have multiple non-maintainer users asking for this behaviour.